code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -234,3 +234,11 @@
 	name = "double bed"
 	desc = "A luxurious double bed, for those too important for small dreams."
 	icon_state = "bed_double"
+	max_buckled_mobs = 2
+
+/obj/structure/bed/double/post_buckle_mob(mob/living/target)
+	if(buckled_mobs.len > 1) //Push the second buckled mob a bit higher from the normal lying position
```

Comment:
I believe with this method you're going to hit this bug:

1. Alice goes into bed, and is at the normal position.
2. Bob goes into bed, at the shifted position.
3. Alice unbuckles from bed, then buckles back into it.
4. Alice and Bob will now both be pixel shifted up.

You'll need to handle this.

Reply:
Done. Thanks for pointing it out.